packages/governance/src/types.js

```diff
@@ -1,77 +1,128 @@
 // @ts-check
 
 /**
- * @typedef { 'amount' | 'brand' | 'installation' | 'instance' | 'nat' | 'ratio' | 'string' | 'unknown' } ParamType
+ * @typedef { 'choose_n' | 'order' | 'weight' } ChoiceMethod
+ *  CHOOSE_N: This is "approval voting". The voter specifies some number of
+ *    positions, and is endorsing them equally.
+ *  ORDER: The voter assigns ordinal numbers to some of the positions. The
+ *    positions will be treated as an ordered list with no gaps.
+ *  WEIGHT: The voter assigns a distinct integer weight to some number of
+ *   positions. The weights are not required to be distinct or consecutive.
+ *   High numbers are most preferred.
  */
 
 /**
- * @typedef { Amount | Brand | Installation | Instance | bigint | Ratio | string | unknown } ParamValue
+ * @typedef { 'param_change' | 'election' | 'survey' } ElectionType
+ * param_change is very specific. Survey means multiple answers are possible,
+ * Election means some candidates are going to "win". It's not clear these are
+ * orthogonal. The important distinction is that param_change has a structured
+ * question, while the others have a question presented as a string.
  */
 
 /**
- * @typedef { 'choose_n' | 'order' | 'weight' } ChoiceMethod
+ * @typedef { 'half' | 'all' | 'none' } QuorumRule
  */
 
 /**
- * @typedef {Object} ParamDescription
- * @property {string} name
- * @property {ParamValue} value
- * @property {ParamType} type
+ * @typedef {Object} SimpleQuestion
+ * @property {string} question
  */
 
 /**
- * @typedef {Object} ParamManagerBase
- * @property {() => Record<Keyword,ParamDescription>} getParams
- *
- * @typedef {{ [updater: string]: (arg: ParamValue) => void }} ParamManagerUpdaters
- * @typedef {ParamManagerBase & ParamManagerUpdaters} ParamManagerFull
+ * @typedef { 'amount' | 'brand' | 'installation' | 'instance' | 'nat' | 'ratio' | 'string' | 'unknown' } ParamType
  */
 
 /**
- * @typedef {Array<ParamDescription>} ParamDescriptions
+ * @typedef { Amount | Brand | Installation | Instance | bigint | Ratio | string | unknown } ParamValue
  */
 
 /**
- * @callback BuildParamManager
- * @param {ParamDescriptions} paramDesc
- * @returns {ParamManagerFull}
+ * @typedef { SimpleQuestion | ParamChangeQuestion } Question
  */
 
 /**
- * @typedef {Object} QuestionTermsShort
- *   BallotDetails as provided to the Registrar
- * @property {ChoiceMethod} method
- * @property {string} question
+ * @typedef {Object} QuestionTerms - BallotSpec plus the Registrar Instance
+ * @property {Question} question
  * @property {string[]} positions
+ * @property {ChoiceMethod} method
  * @property {number} maxChoices
  * @property {ClosingRule} closingRule
+ * @property {Instance} registrar
```

Comment:
If this really is supposed to be a BallotSpec plus Registrar instance, it creates a maintenance hazard to manually repeat all the fields. In fact, checking the two against each other, there are a bunch of differences. I don't know the significance of any of them: Should the `positions` be a `string[]` or a `Position[]`? Why doesn't a `QuestionsTerms` contain `electionType`, `quorumRule`, or `tieOutcome`?

If this really is supposed to be a BallotSpec plus Registrar instance, you could just make it an explicit pair of these things.

```suggestion
 * @typedef {Object} QuestionTerms - BallotSpec plus the Registrar Instance
 * @property {BallotSpec} ballotSpec
 * @property {Instance} registrar
```

Or if you want to treat `QuestionTerms` as an extension/subtype of `BallotSpec`, so that it can include the `BallotSpec` fields directly, you can express subtype as

```suggestion
 * @typedef {BallotSpec} QuestionTerms - BallotSpec plus the Registrar Instance
 * @property {Instance} registrar
```

Reply:
The declaration at point of use got dropped, so lint wasn't telling me about the inconsistencies. I've updated it.